src/Layers/FeatureLayer/FeatureLayer.js

```diff
@@ -89,8 +98,37 @@ EsriLeaflet.Layers.FeatureLayer = EsriLeaflet.Layers.FeatureManager.extend({
         this._map.addLayer(layer);
       }
 
+      // polylines and polygons
       if (layer && layer.setLatLngs) {
-        this._updateLayerGeometry(layer, geojson);
+        this._updateLayer(layer, geojson);
+        // update custom symbology, if necessary
+        this.resetStyle(geojson.id);
```

Comment:
Are you sure you want to `resetStyle ` here? This code just updates the geometry. If I had applied custom styling to an item (like on click) I would not expect it to get reset when I zoomed in.